cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/leveldb/LevelDBTableService.java

```diff
@@ -95,9 +96,16 @@ public void setConfiguration(CConfiguration config) throws IOException {
    * only use in unit test since the singleton may be reused for multiple tests.
    */
   public void clearTables() {
+    for (DB entries : tables.values()) {
+      Closeables.closeQuietly(entries);
+    }
     tables.clear();
   }
 
+  public void close() {
+    clearTables();
```

Comment:
Shouldn't all methods should check if this table service is closed? and if it is closed, throws exception?